From: Robin Cornelius <robin.cornelius@gmail.com>
Subject: ix a crash path in openjpeg where image->comps is wild and gets used in opj_destroy()
Forwarded: no

---
 libopenjpeg/image.c |    1 +
 1 file changed, 1 insertion(+)

--- openjpeg.orig/libopenjpeg/image.c
+++ openjpeg/libopenjpeg/image.c
@@ -28,6 +28,7 @@
 
 opj_image_t* opj_image_create0(void) {
 	opj_image_t *image = (opj_image_t*)opj_calloc(1, sizeof(opj_image_t));
+	image->comps = NULL;
 	return image;
 }
 
